File: ttlnnt/read.mjs
import { Api, JsonRpc, RpcError, JsSignatureProvider } from './dist/index.js'
import * as dotenv from 'dotenv'
dotenv.config()

var api_id = process.argv[2];
const url = process.env.NODE_URL

const json_rpc = new JsonRpc(url) // create new JsonRPC using our node url
const private_key = process.env.PRIVATE_KEY; // private key

const account = process.env.INERY_ACCOUNT // Inery Smart Contract Account to Call
const actor = process.env.INERY_ACCOUNT // The Signer, should match with your provided Private Key
const signature  = new JsSignatureProvider([private_key]); // creating Signer from private key

const api = new Api({
    rpc: json_rpc,
    signatureProvider: signature
})

async function read(id){
    try{
        // create new transaction and sign it
        const tx = await api.transact({
            actions:[
                {
                    account,
                    name:"read",
                    authorization:[
                        {
                            actor,
                            permission:"active"
                        }
                    ],
                    data:{
                        id
                    }
                }
            ]
        },{broadcast:true,sign:true})

        console.log(tx) // output the tx to terminal, it's Json Object
        console.log(tx.processed.action_traces[0].console)
    }catch(error){
        console.log(error)
    }
}

read(api_id)

File: kuroz/read.rb
require 'net/http'
require 'json'

url = "url"
private_key = ENV["PRIVATE_KEY"]
account = "kuroz"
actor = "kuroz"

def perform_transaction(name, data)
    payload = {
        "actions" => [
            {
                "account" => account,
                "name" => name,
                "authorization" => [
                    {
                        "actor" => actor,
                        "permission" => "active"
                    }
                ],
                "data" => data
            }
        ]
    }

    headers = {
        "Content-Type" => "application/json"
    }

    uri = URI(url)
    http = Net::HTTP.new(uri.host, uri.port)
    request = Net::HTTP::Post.new(uri.path, headers)
    request.body = payload.to_json
    response = http.request(request)
    JSON.parse(response.body)
end

def read_transaction(id)
    data = { "id" => id }
    perform_transaction("read", data)
end

def push_transaction(data_id, user, data)
    read_transaction(data_id)
end

push_transaction(1, account, "read transaction")
